src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -142,14 +143,25 @@ def wrapped(func):
 
 class CliCommand(object):
 
-    def __init__(self, name, handler, description=None, table_transformer=None):
+    def __init__(self, name, handler, description=None, table_transformer=None,
+                 arguments_loader=None, description_loader=None):
         self.name = name
         self.handler = handler
-        self.description = description
         self.help = None
+        self.description = description_loader() \
+                           if description_loader and self._should_load_description() \
+                           else description
         self.arguments = {}
+        self.arguments_loader = arguments_loader
         self.table_transformer = table_transformer
 
+    def load_arguments(self):
+        if self.arguments_loader:
+            self.arguments.update(self.arguments_loader())
+
+    def _should_load_description(self):
+        return not APPLICATION.session['completer_active']
```

Comment:
True, but I think you could do that with a table cache:

command                 type
az vm                       group
az vm create            command

Then as part of that pre-screening, if it is a group or a command with help requested, you load the description stuff.

Of course, this only make sense if the time savings justify the work and you are in the best position to know that, not me. :)